R/utils-get_code_dependency.R

```diff
@@ -451,3 +452,61 @@ normalize_pd <- function(pd) {
 
   pd
 }
+
+#' Get line/column in the source where the calls end
+#'
+#'
+#' @param code `character(1)`
+#'
+#' @return `matrix` with `colnames = c("line", "col")`
+#'
+#' @keywords internal
+#' @noRd
+get_call_breaks <- function(code) {
+  parsed_code <- parse(text = code, keep.source = TRUE)
+  pd <- utils::getParseData(parsed_code)
+  pd <- normalize_pd(pd)
+  pd <- pd[pd$token != "';'", ]
+  call_breaks <- t(sapply(
+    extract_calls(pd),
+    function(x) {
+      matrix(c(max(x$line2), max(x$col2)))
+    }
+  ))
+  if (nrow(call_breaks) > 1) call_breaks <- call_breaks[-nrow(call_breaks), ] # breaks in between needed only
+  colnames(call_breaks) <- c("line", "col")
+  call_breaks
+}
+
+#' Split code by calls
+#'
+#' @param code `character` with the code.
+#'
+#' @return list of `character`s of the length equal to the number of calls in `code`.
+#'
+#' @keywords internal
+#' @noRd
+split_code <- function(code) {
+  call_breaks <- get_call_breaks(code)
+  if (nrow(call_breaks) == 1) {
+    return(code)
+  }
+  call_breaks <- call_breaks[order(call_breaks[, "line"], call_breaks[, "col"]), ]
```

Comment:
In case this has one row it gets changed to integer and looses `matrix` class
```suggestion
  call_breaks <- call_breaks[order(call_breaks[, "line"], call_breaks[, "col"]), , drop = TRUE]
```